python3/raygun4py/middleware/django.py

```diff
@@ -1,11 +1,18 @@
 import django
 from django.conf import settings
 from raygun4py import raygunprovider
+from django.utils.functional import cached_property
 
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:
+    MiddlewareMixin = object
 
-class Provider(object):
+class Provider(MiddlewareMixin):
+
+    def __init__(self, get_response=None):
+        super().__init__(get_response)
```

Comment:
If you were to change this line to `self.get_response = get_response` I think the `RecursionError` would go away and the middleware would work.

(I'm not using this client library at the moment, but I'm upgrading a bunch of legacy Django middleware and came across this one and the problem intrigued me.  Feel free to ignore this if it's not helpful).